pkg/k8s/pod_utils.go

```diff
@@ -43,3 +44,45 @@ func LookupContainerPortAndName(pod *corev1.Pod, port intstr.IntOrString, protoc
 	}
 	return 0, "", errors.Errorf("unable to find port %s on pod %s", port.String(), NamespacedName(pod))
 }
+
+// StripDownPodTransformFunc is a transform function that strips down pod to reduce memory usage.
+// see details in [stripDownPodObject].
+func StripDownPodTransformFunc(obj interface{}) (interface{}, error) {
+	if pod, ok := obj.(*corev1.Pod); ok {
+		return stripDownPodObject(pod), nil
+	}
+	return obj, nil
+}
+
+// stripDownPodObject provides an stripDown version of pod to reduce memory usage.
+// NOTE: if the controller needs to refer to more pod fields in the future these fields need to be added to the cache
+func stripDownPodObject(pod *corev1.Pod) *corev1.Pod {
+	pod.ObjectMeta = metav1.ObjectMeta{
+		Name:              pod.Name,
+		Namespace:         pod.Namespace,
+		UID:               pod.UID,
+		DeletionTimestamp: pod.DeletionTimestamp,
+		Labels:            pod.Labels,
+		Annotations:       pod.Annotations,
+		ResourceVersion:   pod.ResourceVersion,
+		Finalizers:        pod.Finalizers,
```

Comment:
Wonder why we need `Annotations` and `Finalizers` ? Annotations could be large as well?